tensordict/tensorclass.py

```diff
@@ -545,19 +558,31 @@ def wrapper(
         **kwargs,
     ):
 
-        for value, key in zip(args, self.__dataclass_fields__):
-            if key in kwargs:
-                raise ValueError(f"The key {key} is already set in kwargs")
-            kwargs[key] = value
+        if not torch.compiler.is_dynamo_compiling():
+            # zip not supported by dynamo
+            for value, key in zip(args, self.__dataclass_fields__):
+                if key in kwargs:
+                    raise ValueError(f"The key {key} is already set in kwargs")
+                kwargs[key] = value
+        else:
+            if args:
+                raise RuntimeError(
+                    "dynamo doesn't support arguments when building a tensorclass, pass the keyword explicitly."
+                )
+
         if batch_size is None:
             batch_size = torch.Size([])
-        for key, field in self.__dataclass_fields__.items():
-            if field.default_factory is not dataclasses.MISSING:
-                default = field.default_factory()
-            else:
-                default = field.default
-            if default not in (None, dataclasses.MISSING):
-                kwargs.setdefault(key, default)
+        if not torch.compiler.is_dynamo_compiling():
+            for key, field in self.__dataclass_fields__.items():
+                if field.default_factory is not dataclasses.MISSING:
+                    default = field.default_factory()
+                else:
+                    default = field.default
+                if default not in (None, dataclasses.MISSING):
+                    kwargs.setdefault(key, default)
+        else:
+            # TODO: Decide what to do here
```

Comment:
@lezcano @anijain2305 

Dynamo doesn't want us to iterate over `self.__dataclass_fields__.items()`: 

```
  File "/Users/vmoens/venv/rl/lib/python3.11/site-packages/torch/_dynamo/symbolic_convert.py", line 1855, in CALL
    self.call_function(fn, args, kwargs)
  File "/Users/vmoens/venv/rl/lib/python3.11/site-packages/torch/_dynamo/symbolic_convert.py", line 739, in call_function
    self.push(fn.call_function(self, args, kwargs))
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vmoens/venv/rl/lib/python3.11/site-packages/torch/_dynamo/variables/misc.py", line 668, in call_function
    return self.obj.call_method(tx, self.name, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vmoens/venv/rl/lib/python3.11/site-packages/torch/_dynamo/variables/misc.py", line 714, in call_method
    return super().call_method(tx, name, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/vmoens/venv/rl/lib/python3.11/site-packages/torch/_dynamo/variables/base.py", line 320, in call_method
    unimplemented(f"call_method {self} {name} {args} {kwargs}")
  File "/Users/vmoens/venv/rl/lib/python3.11/site-packages/torch/_dynamo/exc.py", line 216, in unimplemented
    raise Unsupported(msg)
torch._dynamo.exc.Unsupported: call_method GetAttrVariable(UserDefinedObjectVariable(MyClass), __dataclass_fields__) items [] {}

```